crypto/bls12381/key_bls12381.go

```diff
@@ -0,0 +1,159 @@
+//go:build ((linux && amd64) || (linux && arm64) || (darwin && amd64) || (darwin && arm64) || (windows && amd64)) && bls12381
+
+package bls12381
+
+import (
+	"bytes"
+
+	sha256 "github.com/minio/sha256-simd"
+
+	"github.com/cometbft/cometbft/crypto"
+	bls12381 "github.com/cosmos/crypto/curves/bls12381"
+
+	"github.com/cometbft/cometbft/crypto/tmhash"
+	cmtjson "github.com/cometbft/cometbft/libs/json"
+)
+
+const (
+	// PrivKeySize defines the length of the PrivKey byte array.
+	PrivKeySize = 32
+	// PubKeySize defines the length of the PubKey byte array.
+	PubKeySize = 48
+	// SignatureLength defines the byte length of a BLS signature.
+	SignatureLength = 96
+	// KeyType is the string constant for the bls12_381 algorithm.
+	KeyType = "bls12_381"
+	// Enabled indicates if this curve is enabled.
+	Enabled = true
+)
+
+// -------------------------------------.
+const (
+	PrivKeyName = "cometbft/PrivKeyBLS12_381"
+	PubKeyName  = "cometbft/PubKeyBLS12_381"
+)
+
+func init() {
+	cmtjson.RegisterType(PubKey{}, PubKeyName)
+	cmtjson.RegisterType(PrivKey{}, PrivKeyName)
+}
+
+// ===============================================================================================
+// Private Key
+// ===============================================================================================
+
+// PrivKey is a wrapper around the Ethereum bls12_381 private key type. This
+// wrapper conforms to crypto.Pubkey to allow for the use of the Ethereum
+// bls12_381 private key type.
+
+var _ crypto.PrivKey = &PrivKey{}
+
+type PrivKey []byte
+
+func NewPrivateKeyFromBytes(bz []byte) (PrivKey, error) {
+	secretKey, err := bls12381.SecretKeyFromBytes(bz)
+	if err != nil {
+		return nil, err
+	}
+	return secretKey.Marshal(), nil
+}
+
+func GenPrivKey() (PrivKey, error) {
+	secretKey, err := bls12381.RandKey()
+	return PrivKey(secretKey.Marshal()), err
+}
+
+// Bytes returns the byte representation of the Private Key.
+func (privKey PrivKey) Bytes() []byte {
+	return privKey
+}
+
+// PubKey returns the private key's public key. If the privkey is not valid
+// it returns a nil value.
+func (privKey PrivKey) PubKey() crypto.PubKey {
+	secretKey, _ := bls12381.SecretKeyFromBytes(privKey)
```

Comment:
Please, use the error value here.
This unit test panics on line 76:
```go
func TestInvalidPrivKeyBLS12381(t *testing.T) {
	privKey, err := bls12381.GenPrivKey()
	require.NoError(t, err)
	privKey = privKey[1:] // corrupt key
	pubKey := privKey.PubKey()
	require.Nil(t, pubKey)
}
```

Reply:
Good catch 👍